### ISSUE TITLE
flag parsing terminator is broken

### ISSUE THREAD

**Issue by philips** [codegangsta/cli#56]
I want to be able to use the classic `--` command parsing terminator in an app using the cli package. However, something strange happens:

```
./example  --lang english echo  foobar  python myscript.py
foobar [python myscript.py]
```
```
./example  --lang english echo  foobar -- python myscript.py -d
python [myscript.py -d foobar]
```

foobar is jumping from the 0th arg to the 2nd arg if the `--` terminator is given! I haven't started digging into the code but this is rather suprising.

```
package main

import (
	"fmt"
	"os"

	"github.com/codegangsta/cli"
)

func main() {
	app := cli.NewApp()
	app.Flags = []cli.Flag {
		cli.StringFlag{"lang, l", "english", "language for the greeting"},
	}
	app.Commands = []cli.Command{
		{
			Name: "echo",
			Flags: []cli.Flag{
				cli.StringFlag{"newline, n", "yes", "print newline"},
			},

			Action: func(c *cli.Context) {
				args := c.Args()
				println(args[0], fmt.Sprintf("%v", args[1:]))
			},
		},
	}

	app.Run(os.Args)
}
```

**Comment by jszwedko** [comment]
Fixed by #157.

**Comment by philips** [comment]
@jszwedko Awesome, thanks.
